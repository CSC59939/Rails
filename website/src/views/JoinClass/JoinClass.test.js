import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Button, Card, Icon } from 'antd';
import { JoinClass } from '..';
import '../../utils/tests/test.css';
 configure({ adapter: new Adapter() });
 describe("JoinClass Page", () => {
  it('Render JoinClass page without any error', () => { // eslint-disable-line no-undef
    expect(renderer.create(<JoinClass />).toJSON()).toMatchSnapshot();
  });
})
