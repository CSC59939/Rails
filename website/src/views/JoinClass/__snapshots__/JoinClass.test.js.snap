// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinClass Page Render JoinClass page without any error 1`] = `
<div
  className="join"
>
  <h1
    className="title"
  >
    Rails
  </h1>
  <div
    className="ant-card joinclasscard ant-card-bordered"
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          Join Class
        </div>
        <div
          className="ant-card-extra"
        >
          <a
            href="#"
          />
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <form
        className="ant-form ant-form-horizontal regiform"
        onSubmit={[Function]}
      >
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-24"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <div
                  className="ant-select ant-select-enabled"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                >
                  <div
                    aria-autocomplete="list"
                    aria-expanded={false}
                    aria-haspopup="true"
                    className="ant-select-selection
            ant-select-selection--single"
                    onKeyDown={[Function]}
                    role="combobox"
                    tabIndex={0}
                  >
                    <div
                      className="ant-select-selection__rendered"
                    >
                      <div
                        className="ant-select-selection__placeholder"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        style={
                          Object {
                            "WebkitUserSelect": "none",
                            "display": "block",
                            "userSelect": "none",
                          }
                        }
                        unselectable="on"
                      >
                        University/College
                      </div>
                      <div
                        className="ant-select-search ant-select-search--inline"
                        style={
                          Object {
                            "display": "none",
                          }
                        }
                      >
                        <div
                          className="ant-select-search__field__wrap"
                        >
                          <input
                            autoComplete="off"
                            className="ant-select-search__field"
                            onChange={[Function]}
                            onKeyDown={[Function]}
                            value=""
                          />
                          <span
                            className="ant-select-search__field__mirror"
                          >
                            
                             
                          </span>
                        </div>
                      </div>
                    </div>
                    <span
                      className="ant-select-arrow"
                      onClick={[Function]}
                      style={
                        Object {
                          "WebkitUserSelect": "none",
                          "userSelect": "none",
                        }
                      }
                      unselectable="on"
                    >
                      <i
                        className="anticon anticon-down ant-select-arrow-icon"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="down"
                          fill="currentColor"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </i>
                    </span>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-24"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <div
                  className="ant-select ant-select-enabled"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                >
                  <div
                    aria-autocomplete="list"
                    aria-expanded={false}
                    aria-haspopup="true"
                    className="ant-select-selection
            ant-select-selection--multiple"
                    role="combobox"
                  >
                    <div
                      className="ant-select-selection__rendered"
                    >
                      <div
                        className="ant-select-selection__placeholder"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        style={
                          Object {
                            "WebkitUserSelect": "none",
                            "display": "block",
                            "userSelect": "none",
                          }
                        }
                        unselectable="on"
                      >
                        Class and Section code
                      </div>
                      <ul>
                        <li
                          className="ant-select-search ant-select-search--inline"
                        >
                          <div
                            className="ant-select-search__field__wrap"
                          >
                            <input
                              autoComplete="off"
                              className="ant-select-search__field"
                              onChange={[Function]}
                              onKeyDown={[Function]}
                              value=""
                            />
                            <span
                              className="ant-select-search__field__mirror"
                            >
                              
                               
                            </span>
                          </div>
                        </li>
                      </ul>
                    </div>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
        <div
          className="registerButton"
        >
          <button
            className="ant-btn ant-btn-primary"
            margin="auto"
            onClick={[Function]}
            type="submit"
          >
            <span>
              Join
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
  <br />
  <div
    className="ant-card request_permission ant-card-bordered"
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          Request Permission
        </div>
        <div
          className="ant-card-extra"
        >
          <a
            href="#"
          />
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <form
        className="ant-form ant-form-horizontal permission_form"
        onSubmit={[Function]}
      >
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-24"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <div
                  className="ant-select ant-select-enabled"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                >
                  <div
                    aria-autocomplete="list"
                    aria-expanded={false}
                    aria-haspopup="true"
                    className="ant-select-selection
            ant-select-selection--multiple"
                    role="combobox"
                  >
                    <div
                      className="ant-select-selection__rendered"
                    >
                      <div
                        className="ant-select-selection__placeholder"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        style={
                          Object {
                            "WebkitUserSelect": "none",
                            "display": "block",
                            "userSelect": "none",
                          }
                        }
                        unselectable="on"
                      >
                        Class and Section code
                      </div>
                      <ul>
                        <li
                          className="ant-select-search ant-select-search--inline"
                        >
                          <div
                            className="ant-select-search__field__wrap"
                          >
                            <input
                              autoComplete="off"
                              className="ant-select-search__field"
                              onChange={[Function]}
                              onKeyDown={[Function]}
                              value=""
                            />
                            <span
                              className="ant-select-search__field__mirror"
                            >
                              
                               
                            </span>
                          </div>
                        </li>
                      </ul>
                    </div>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-24"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <input
                  className="ant-input"
                  disabled={false}
                  onKeyDown={[Function]}
                  placeholder="Student ID #"
                  type="text"
                />
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-24"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <input
                  className="ant-input"
                  disabled={false}
                  onKeyDown={[Function]}
                  placeholder="College Email"
                  type="text"
                />
              </span>
            </div>
          </div>
        </div>
        <div
          className="registerButton"
        >
          <button
            className="ant-btn ant-btn-primary"
            margin="auto"
            onClick={[Function]}
            type="submit"
          >
            <span>
              Request
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
